### ISSUE TITLE
`equals` and recursive structures leads to `Maximum call stack size exceeded`

### ISSUE THREAD

**Issue by dartess** [remeda/remeda#351]
```js
let c = {}; c.v = c;
let d = {}; d.v = d;
console.log(R.equals(c, d));
```

```
Maximum call stack size exceeded
```

**Comment by eranhirsch** [comment]
I'd say this is an edge case that is not worth adding more complexity (and runtime cost) for all invocations. I don't mind adding a `equals.recursive` flavor of equals so that users can opt-in to it, but I don't think it will see much usage.

@TkDodo, wdyt?

**Comment by dartess** [comment]
I agree. But if we decide that this is "an edge case that is not worth adding more complexity (and runtime cost)", I think the recursive checks can also be removed from [PR with mergeDeep](https://github.com/remeda/remeda/pull/317).

**Comment by eranhirsch** [comment]
I generally agree, with the onlh difference being that that is a new function without existing users so new adopters take it as a whole. Fixing equals would mean changing it for existing users

**Comment by eranhirsch** [comment]
@dartess, is there anything actionable you want to do with this issue? I'm thinking of closing it as it's now findable if anyone else encounters the issue and search our issues so they can find this discussion.

**Comment by dartess** [comment]
I don't think it's necessary to do something about it if you don't feel like doing something about it. So, just closing.

**Comment by filipw01** [comment]
When trying to migrate from `lodash` I encountered the same issue (function was renamed to `isDeepEqual`). Should we do something about it? Is there any way to handle this at the moment? If fixing it is not the way, could we add a third argument with options and add a `recursive` option?

**Comment by filipw01** [comment]
I know it's generally a bad practice, but my example is passing `isDeepEqual` as [`arePropsEqual` in `React.memo`](https://react.dev/reference/react/memo#parameters)
